src/error.rs

```diff
@@ -1,16 +1,30 @@
 use std::{num, str, string};
 
+/// Represents all of the errors that can arise when creating, deserializing, or verifying macaroons.
+///
+/// `InitializationError` is only raised by the [`initialize`] function, when there is a problem
+/// initializing the lower-level crypo library. `CryptoError` represents a runtime error when using
+/// that library, or situations like zero-length ciphertext.
+///
+/// `IncompleteMacaroon` and `IncompleteCaveat` can occur when constructing or deserializing
+/// [`Macaroon`] or [`Caveat`] structs, and expected fields are not present.
+///
+/// `DeserializationError` represents a broad category of issues when parsing a macaroon token in
+/// any format.
+///
+/// `CaveatNotSatisfied`, `DischargeNotUsed`, and `InvalidSignature` are all errors that arise when
+/// verifying a [`Macaroon`], and all represent a failure to authorize with the given key and set
+/// of satisfiers.
 #[derive(Debug)]
 pub enum MacaroonError {
```

Comment:
I'm not sure of your preferences but I think it might be better to have these docs inline for each variant. I see there would be some duplication which is meh, but I think it's more discoverable/common. Open to suggestions of course!

Reply:
Ah! Forgot that I could do that and it would render well in the docs, I updated to that style.